File: src/Services/Review/Review.Application/DTOs/ResponseDTOs/ReviewResponseDto.cs
using Review.Domain.Entities;

namespace Review.Application.DTOs.ResponseDTOs;

public sealed record ReviewResponseDto(
    string Id,
    int BookId,
    int UserId,
    int Rating,
    string? Text,
    IEnumerable<int> LikeUserIds,
    IEnumerable<int> DislikeUserIds,
    IEnumerable<Comment>? Comments,
    UserResponseDto User,
    BookResponseDto Book);

File: src/Client/Angular.Client/src/app/models/review/reviewResponseDto.ts
import { BookResponseDto } from './bookResponseDto';
import { Comment } from './comment';
import { UserResponseDto } from './userResponseDto';

export interface ReviewResponseDto {
    id: string;
    bookId: number;
    userId: number;
    rating: number;
    text: string;
    likeUserIds: Array<number>;
    dislikeUserIds: Array<number>;
    comments: Array<Comment>;
    user: UserResponseDto;
    book: BookResponseDto;
}
